src/applications/mhv-medical-records/util/helpers.js

```diff
@@ -430,6 +430,47 @@ export const radiologyRecordHash = async record => {
   return (await generateHash(dataString)).substring(0, 8);
 };
 
+/**
+ * @param {Object} nameObject {first, middle, last, suffix}
+ * @returns {String} formatted timestamp
+ */
+
+export const formatNameFirstLast = ({ first, middle, last, suffix }) => {
+  let returnName = '';
+  if (!last) return null;
+
+  let firstName = `${first}`;
+  let lastName = `${last}`;
+
+  if (!first) {
+    return lastName;
+  }
+  if (middle) firstName += ` ${middle}`;
+  if (suffix) lastName += `, ${suffix}`;
+
+  returnName = `${firstName} ${lastName}`;
+
+  return returnName;
+};
+
+/**
+ * @param {Object} nameObject name
+ * @returns {String} formatted timestamp
+ */
+
+export const formatNameFirstToLast = name => {
+  try {
+    const parts = name?.split(',');
+    if (parts.length !== 2) {
+      return null;
+    }
+    const [lastname, firstname] = parts;
+    return `${firstname} ${lastname}`;
+  } catch {
+    return null;
+  }
+};
+
```

Comment:
This function splits on a comma in the name text. Have you verified that this data will always be in the form "Lastname, Firstname"? If not, maybe we can check for the existence of the comma before formatting. Also check for strange unhandle-able cases like multiple commas.

I know some fields are split out in FHIR by various name types. Did none of our names come from those structures where we can just format them as we choose? e.g.
```
"name": [
  { "text": "SMITH,JOHN", "family": "JOHN", "given": ["SMITH"] }
]
```

Finally, I don't think you should return null here. I would think if everything else fails you should return the original name.

Reply:
Every record with a "name" key doesn't contain "family" and "given" keys within them. But they all have "text". But some of the "text" keys have "firstName,lastName" instead of "lastName,firstName" for the same field in the array.

Reply:
I think tomorrow at the dev meetup we can discuss an approach to how we're going to do this I cannot see a one-size-fits-all approach considering the data we are getting.